test_run.lua

```diff
@@ -284,6 +285,39 @@ local function grep_log(self, node, what, bytes, opts)
     return found
 end
 
+-- Checks cmp(fn()) with delay until timeout or returns fn().
+-- Comparator cmp should return true if it what we waiting for.
+local function wait_cond(self, fn, timeout, delay, cmp)
```

Comment:
I think `cmp` is redundant here, we can just return what `fn` gives us and implement some complex behavious with custom `fn`. This way we leave the common code easy to read, while do not lost its powerfulness.